contracts/chain-adapters/Scroll_Adapter.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@scroll-tech/contracts/L1/gateways/IL1GatewayRouter.sol";
+import "@scroll-tech/contracts/L1/IL1ScrollMessenger.sol";
+import "./interfaces/AdapterInterface.sol";
+
+contract Scroll_Adapter is AdapterInterface {
+    using SafeERC20 for IERC20;
+    uint32 public immutable l2GasLimit = 250_000;
+
+    IL1GatewayRouter public immutable l1GatewayRouter;
+    IL1ScrollMessenger public immutable l1ScrollMessenger;
+
+    /**
+     * @notice Constructs new Adapter.
+     * @param _l1GatewayRouter Standard bridge contract.
+     * @param _l1ScrollMessenger Scroll Messenger contract.
+     */
+    constructor(IL1GatewayRouter _l1GatewayRouter, IL1ScrollMessenger _l1ScrollMessenger) {
+        l1GatewayRouter = _l1GatewayRouter;
+        l1ScrollMessenger = _l1ScrollMessenger;
+    }
+
+    /**
+     * @notice Send message to `target` on L2.
+     * @dev This method is marked payable because relaying the message might require a fee
+     * to be paid by the sender to forward the message to L2. However, it will not send msg.value
+     * to the target contract on L2.
+     * @param target L2 address to send message to.
+     * @param message Message to send to `target`.
+     */
+    function relayMessage(address target, bytes calldata message) external payable {
+        // We can specifically send a message with 0 value to the Scroll Bridge
+        // and it will not forward any ETH to the target contract on L2. However,
+        // we need to set the payable value to msg.value to ensure that the Scroll
+        // Bridge has enough gas to forward the message to L2.
+        l1ScrollMessenger.sendMessage{ value: msg.value }(target, 0, message, l2GasLimit);
```

Comment:
Does the `msg.value` have to account in any way for the l2 gas cost? Also, does this contract have to hold ETH to pay for the message like the `arbitrum_adapter` does or does it do something more like the `ovm_adapter` where the L2 recipient pays for the gas

Reply:
for reference, when executing pool rebalance leaves we need to send ETH to the HubPool: https://github.com/across-protocol/relayer-v2/blob/master/src/dataworker/Dataworker.ts#L1828

Reply:
Similar comment: https://github.com/across-protocol/contracts-v2/pull/382#discussion_r1433463127.

We probably don't want to require the executor to send funds with the call to the HubPool.

Reply:
Agreed, but we may have to do something like in Arbitrum_Adapter where we assume the HubPool has a balance of ETH from which to draw its msg.value from

Reply:
Yep - the new implementation will need to call `loadEthForL2Calls` but `msg.value` is no longer relied upon